specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json

```diff
@@ -12605,17 +12719,88 @@
           "$ref": "#/definitions/WindowsConfiguration",
           "title": "Windows operating system settings on the virtual machine.",
           "description": "This property must not be specified if the imageReference or osDisk property specifies a Linux OS image."
+        },
+        "dataDisks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataDisk"
+          },
+          "title": "The configuration for data disks attached to the comptue nodes in the pool.",
+          "description": "This property must be specified if the compute nodes in the pool need to have empty data disks attached to them. This cannot be updated."
+        },
+        "licenseType": {
+          "type": "string",
+          "title": "If the the image or disk that is being used was licensed on-premises.",
+          "description": "This only applies to images that contain the Windows operating system. Values are:\n\n Windows_Server - The on-premises license is for Windows Server.\n Windows_Client - The on-premises license is for Windows Client."
+        },
+        "containerConfiguration": {
+          "$ref": "#/definitions/ContainerConfiguration",
+          "title": "The container configuration for the pool.",
+          "description": "If this is specified, then all regular tasks and job manager tasks run on this pool must specify the containerSettings property."
         }
       },
-      "required": [ "nodeAgentSKUId" ],
+      "required": [ "nodeAgentSKUId", "imageReference" ],
```

Comment:
It wasn't required before because customImage was on a different property, and it was an "pick 1" situation. But now that custom image stuff is IN the image reference, we can make it required (which is actually nicer, so I'm happy)